crates/bevy_ecs/src/name.rs

```diff
@@ -213,10 +221,44 @@ impl Deref for Name {
     }
 }
 
+#[cfg(feature = "serialize")]
+impl Serialize for Name {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        serializer.serialize_str(self.as_str())
+    }
+}
+
+#[cfg(feature = "serialize")]
+impl<'de> Deserialize<'de> for Name {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        deserializer.deserialize_str(EntityVisitor)
+    }
+}
+
+#[cfg(feature = "serialize")]
+struct EntityVisitor;
```

Comment:
Could we give a better name for this while we're at it?

Reply:
Yeah I agree, it's a bad name. I've chosen `NameVisitor`, since it's a visitor for the `Name` component.